src/app/calculator/utilities/altitude-correction/altitude-correction.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { BehaviorSubject } from 'rxjs';
+import { ConvertUnitsService } from '../../../shared/convert-units/convert-units.service';
+import { Settings } from '../../../shared/models/settings';
+
+@Injectable()
+export class AltitudeCorrectionService {
+
+  currentField: BehaviorSubject<string>;
+  resetData: BehaviorSubject<boolean>;
+  generateExample: BehaviorSubject<boolean>;
+
+  altitudeCorrectionInputs: BehaviorSubject<number>; 
+  altitudeCorrectionOutputs: BehaviorSubject<number>;
+
+  constructor(private convertUnitsService: ConvertUnitsService, private formBuilder: FormBuilder) {
+    this.currentField = new BehaviorSubject<string>('default');
+    this.resetData = new BehaviorSubject<boolean>(undefined);
+    this.generateExample = new BehaviorSubject<boolean>(undefined);
+    this.altitudeCorrectionInputs = new BehaviorSubject<number>(undefined);
+    this.altitudeCorrectionOutputs = new BehaviorSubject<number>(undefined);
+  }
+
+  getFormFromObj(obj: number): FormGroup {
+    let form = this.formBuilder.group({
+      altitude: [obj, Validators.required]
+    });
+    return form;
+  }
+
+  getObjFromForm(form: FormGroup): number {
+    let altitude: number = form.controls.altitude.value;
+    return altitude;
+  }
+
+  getDefaultData(settings: Settings) {
+    let altitude = 0;
+    if (settings.unitsOfMeasure == 'Metric') {
```

Comment:
This block to check for unit setting can be removed. Initializing empty input here so it's always 0 right?